samples/algorithms/reversible-logic-synthesis/ReversibleLogicSynthesis.qs

```diff
@@ -350,11 +108,11 @@ namespace Microsoft.Quantum.Samples.ReversibleLogicSynthesis {
     ///    2010](https://doi.org/10.1137/1.9781611973075.37)
     operation FindHiddenShift (perm : Int[], shift : Int) : Int {
 
-        let n = BitSizeI(Length(perm));
+        let n = BitSizeI(Length(perm) - 1);
         using (qubits = Qubit[2 * n]) {
             let Superpos = ApplyToEachA(H, _);
             let Shift = ApplyShift(shift, _);
-            let Synth = ApplyPermutationOracle(perm, TBS, _);
+            let Synth = LittleEndianWrapper(ApplyPermutationUsingTransformation(perm, _), _);
             let PermX = ApplyToSubregisterA(Synth, RangeAsIntArray(0..n - 1), _);
             let PermY = ApplyToSubregisterA(Synth, RangeAsIntArray(n..2 * n - 1), _);
             ApplyWith(BoundA([Superpos, Shift, PermY]), ComputeInnerProduct, qubits);
```

Comment:
As a side thought, it may be simpler to rewrite this using the new `within`/`apply` feature than using `LittleEndianWrapper` to force things into the type signature of BoundA.

Reply:
Yes, that makes the code so much simpler. Thanks